aptos-move/aptos-gas-schedule/src/gas_schedule/misc.rs

```diff
@@ -304,6 +314,12 @@ impl AbstractValueSizeGasParameters {
         }
 
         impl<'a> ValueVisitor for Visitor<'a> {
+            #[inline]
+            fn visit_delayed(&mut self, _depth: usize, _val: DelayedValueHandle) {
```

Comment:
I didn't understand the concept of `ValueVisitor` trait. For all these trait functions, we specify a depth and a value. The comments say that these functions are used to travel the value recursively. We can't traverse simple types like u8, u32, etc. So, I didn't understand the use of functions like visit_u8, visit_u32, etc. In this context, we seem to be returning the gas parameters for the type. What's the general usage for ValueVisitor?

Reply:
Yes, u8, etc. are leaf nodes of the tree (`Value`) but you still may want to do certain things when visiting them. For example, you can have a visitor which counts how many `U8(100)` are there. You can check the implementation of the visitor which the gas meter is using for calculating the heap/stack usage - we increment counters based on the sizes of visited values, including leaf nodes such as integers.